### ISSUE TITLE
Rails 5.0.0.beta - 204 no content confusing in development

### ISSUE THREAD

**Issue by whistlerbrk** [rails/rails#23077]
When a controllers method is reached but the template can not be found, Rails now returns 204 No Content.

As a result, the browser does not refresh but more important - the current page whatever it was, does not clear an error if one is being displayed. For a few moments I thought I'd run into my prior issue.

Now, I experienced this when I renamed a controller and a route but forgot to move the views directory, a naive thing. In previous version of Rails this would have tried to render and failed, and I'm sure this new behaviour is here for a reason. However, I think some here may see how this could momentarily cause confusion -- would blanking out the current error if one is shown here a good idea? This would at least then cause someone to turn to their log to see what is happening.

**Comment by maclover7** [comment]
Yep, this is being worked on as apart of #20666 and #23062.

**Comment by matthewd** [comment]
@maclover7 those issues are about a different condition under which we might choose not to 204.

This is more along the lines of the discussion around https://github.com/rails/rails/issues/19036#issuecomment-171000017.

**Comment by clarkware** [comment]
I agree that the current behavior is very confusing, and not just for beginners. I've been using Rails for 10 years now, and still got tripped up by a missing template the first time I started playing with Rails 5. At first I thought it was a Rails bug since the previous error message wasn't being cleared from the browser. It was only after digging into the log file that I noticed the 204 response. 

For HTML apps, it's super helpful to see the familiar "Missing template" error in the browser in the same way that it's helpful to see "Unknown action" and other errors in the browser. I understand that a 204 is sufficient for API controllers, but it would be a shame if Rails 5 shipped with no visual feedback in the browser when an expected template is missing.

Can anyone confirm whether this is being worked on?
